code/datums/components/embedded.dm

```diff
@@ -287,3 +288,28 @@
 	to_chat(user, span_notice("You successfully pluck [weapon] from [victim]'s [limb.plaintext_zone]."))
 	to_chat(victim, span_notice("[user] plucks [weapon] from your [limb.plaintext_zone]."))
 	safeRemove(user)
+
+/// Called when an object is ripped out of someone's body by magic or other abnormal means
+/datum/component/embedded/proc/magic_pull(datum/source, mob/living/caster, obj/marked_item)
+	SIGNAL_HANDLER
+
+	if(marked_item != weapon || src.limb != limb)
```

Comment:
``limb`` isn't defined as anything so it's still src.limb, this is just checking if(limb != limb)